proposals/NativeInt.md

```diff
@@ -0,0 +1,264 @@
+# Native-sized integers
+
+## Summary
+[summary]: #summary
+
+Language support for a native-sized signed and unsigned integer types.
+
+The motivation is for interop scenarios and for low-level libraries.
+
+## Design
+[design]: #design
+
+New contextual keywords `nint` and `nuint` represent native signed and unsigned integer types.
+The identifiers are only treated as keywords when used in a type context and when the identifier does not otherwise bind to a symbol at that program location.
+
+The types `nint` and `nuint` are essentially aliases to underlying types `System.IntPtr` and `System.UIntPtr`, where the compiler can surface additional conversions and operations for native ints.
+
+### Constants
+
+There is no direct syntax for native int literals. Explicit casts of other integral constant values can be used instead: `(nint)42`.
+
+`nint` constants are in the range [ `int.MinValue`, `int.MaxValue` ].
+
+`nuint` constants are in the range [ `uint.MinValue`, `uint.MaxValue` ].
+
+There are no `MinValue` or `MaxValue` fields on `nint` or `nuint` because, other than `nuint.MinValue`, those values cannot be emitted as constants.
+
+Constant folding is supported for all operators: { (unary)`-`, `~`, `+`, `-`, `*`, `/`, `%`, `==`, `!=`, `<`, `<=`, `>`, `>=`, `&`, `|`, `<<`, `>>` }.
+Constant folding operations are evaluated with `Int32` and `UInt32` operands rather than native ints for consistent behavior regardless of compiler platform.
+
+### Conversions
+Types that differ only by `nint` and `IntPtr` and by `nuint` and `UIntPtr` are considered equivalent. That applies to the primitive types as well as arrays, `Nullable<>`, constructed types, and tuples.
+
+_Are these "identity" conversions between equivalent types represented in the Roslyn API?_
+
+The following numeric conversions are supported.
```

Comment:
>The following numeric conversions are supported [](start = 0, length = 47)

Please (also) describe as an adjustment to [https://github.com/dotnet/csharplang/blob/master/spec/conversions.md#implicit-numeric-conversions](https://github.com/dotnet/csharplang/blob/master/spec/conversions.md#implicit-numeric-conversions) and [https://github.com/dotnet/csharplang/blob/master/spec/conversions.md#explicit-numeric-conversions](https://github.com/dotnet/csharplang/blob/master/spec/conversions.md#explicit-numeric-conversions).  Include the conversion classifications please (identity, implicit numeric, explicit numeric, enum conversion, etc).

Please say if any other conversions exist.  For example, is there an explicit conversion from an enum type to nint?